### PR TITLE
feat: make projectsFor a React component

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Tested on `/settings`, and `/certification/certifieduser/legacy-front-end` routes.

### CODE REVIEW

File: client/src/client-only-routes/show-project-links.tsx [hunk 1]
```diff
@@ -113,33 +113,41 @@ const ShowProjectLinks = (props: ShowProjectLinksProps): JSX.Element => {
...
+            );
+          })}
+        </>
+      );
+    }
+
+    const project = certsToProjects[certName];
+    return (
+      <>
+        {project.map(({ link, title, id }) => (
```
Comment (Sboonny) [inline client/src/client-only-routes/show-project-links.tsx:139]:
```suggestion
    const projects = certsToProjects[certName];
    return (
      <>
        {projects.map(({ link, title, id }) => (
```

File: client/src/client-only-routes/show-project-links.tsx [hunk 2]
```diff
@@ -113,33 +113,41 @@ const ShowProjectLinks = (props: ShowProjectLinksProps): JSX.Element => {
...
+          <tr key={id}>
             <td>
-              <Link className='project-link' to={certLocation} external>
-                {t(`certification.title.${cert.title}`, cert.title)}
+              <Link to={link}>
+                {t(`certification.project.title.${title}`, title)}
```
Comment (Sboonny) [inline client/src/client-only-routes/show-project-links.tsx:143]:
```suggestion
                {t(`certification.projects.title.${title}`, title)}
```

**Review by Sboonny — COMMENTED** [review]
renamed the variable because it should be an array of elements

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**